### PR TITLE
Setup CI with CircleCI and Quay

### LINKED ISSUE #8: Set Up Continuous Integration Server
We need to set up continuous integration somewhere - this is going to be open source so we can rely on an open source solution like CircleCI if we would like. Requirements are going to be pretty minimal

 - Run `./scripts/tests`
 - Run `./scripts/cipublish` - which should publish an application container to a public container repository

### PR DESCRIPTION
**Pull request by rbreslow**
## Overview

- Add `.circleci/config.yml` file
- Add README badges 😎
- Use Scala 2.12 artifacts in `granary-api` container image
- Fix `scapegoatVersion` scope
- Add `app-backend/database/src/main/resources/db/.gitkeep` so `granary-api-migrations` can build
- Use Paul Phillip's more capable [sbt runner](https://github.com/paulp/sbt-extras)
- Add `shellcheck` Docker Compose service
- Add `cipublish` to publish container images to [Quay.io](https://quay.io/organization/raster-foundry)
- Add mechanism for generating `.sbtopts` to resolve CI build failures
- Align impacted STRTA with the [ADR](https://github.com/azavea/architecture/blob/master/doc/arch/adr-0000-scripts-to-rule-them-all.md) and Google's [Shell Style Guide](https://google.github.io/styleguide/shell.xml)

Resolves #8 

### Notes

I wanted to get Slack notifications wired up. It seems there are two paths we could go down. CircleCI lets you [configure notifications](https://circleci.com/docs/2.0/notifications/#enable-chat-notifications) per project, but there's no way to filter based on branch, so this would be information overload. There's also the [Slack orb](https://github.com/CircleCI-Public/slack-orb). This option seemed equally inflexible. We wouldn't be able to gate against PRs or have notification for fixes. Also, it requires using the 2.1 config version, which seems to break things. From what I can tell, it's [still in alpha](https://discuss.circleci.com/t/2-1-config-and-build-processing/24102/11).

Also, I wanted to split out `cipublish` execution into a `deploy` job, but there's no way to share the Docker execution environment across jobs. Here's a related [thread](https://discuss.circleci.com/t/can-docker-images-be-preserved-between-jobs-in-a-workflow-without-a-manual-load-save/23388/11).

## Testing Instructions

- See CircleCI build: https://circleci.com/gh/raster-foundry/granary/21
- See container images on Quay.io: https://quay.io/repository/raster-foundry/granary-api

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      # https://circleci.com/docs/2.0/building-docker-images/
+      - setup_remote_docker:
+          docker_layer_caching: true
+
+      - restore_cache:
+          keys:
+            - sbt-cache-{{ checksum "app-backend/project/Dependencies.scala" }}
+
+      - run:
+          command: sbt test:compile
```
Comment (rbreslow) [inline .circleci/config.yml]: Split out fetching dependencies from the rest of the build for more efficient caching.

File: app-backend/build.sbt [hunk 2]
```diff
@@ -11,7 +11,7 @@ lazy val commonSettings = Seq(
   version := "0.0.1-SNAPSHOT",
   scalaVersion := "2.12.8",
   scalafmtOnCompile := true,
-  scapegoatVersion := Versions.ScapegoatVersion,
+  scapegoatVersion in ThisBuild := Versions.ScapegoatVersion,
```
Comment (rbreslow) [inline app-backend/build.sbt]: Not too satisfied with the conclusion of throwing `in ThisBuild` at it and calling it a day.

We're adding `commonSettings` to each project below, so this shouldn't be needed?

File: scripts/test [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+"
+}
+
+if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
+    if [ "${1:-}" = "--help" ]; then
+        usage
+    else
+        echo "Linting Bash scripts"
+        if [[ $CI ]]; then
+            # create a dummy container which will hold a volume with our sources
```
Comment (rbreslow) [inline scripts/test]: >It is **not** possible to mount a folder from your job space into a container in Remote Docker (and vice versa)

See: https://circleci.com/docs/2.0/building-docker-images/#mounting-folders

File: app-backend/build.sbt (hunk 2 above)
Comment (notthatbreezy) [reply app-backend/build.sbt]: I wouldn't sweat this -- I agree that we shouldn't need it, but I don't think there's a good reason to worry about it.

**Comment by rbreslow** [comment]
@notthatbreezy I wired up Slack notifications with `curl`. Let me know what you think.

Also, I was using `circleci-agent step halt` before, but it doesn't look like it was actually preventing `cipublish` from running. It basically says *after* this step, don't run any more steps.

So, now, I'm wrapping `cipublish` in a conditional. This also means we can reach the Slack notifications in case of failure.

**Comment by notthatbreezy** [comment]
@rbreslow - slack notifications look great!

**Pull request merged**